logger/src/main/java/com/orhanobut/logger/FileLogger.java

```diff
@@ -0,0 +1,362 @@
+         */
+        public Builder logLevel(int logLevel) {
+            this.logLevel = logLevel;
+            return this;
+        }
+
+        /**
+         * After maximum number of lines is reached a new file is created.
+         *
+         * @param maxLines maximum number of lines per log file
+         */
+        public Builder maxLines(int maxLines) {
+            this.maxLines = maxLines;
+            return this;
+        }
+
+        /**
+         * Path to store logs relative to Environment.getExternalStorageDirectory()
+         * NEEDS STORAGE PERMISSION!
+         *
+         * @param folder folder name
+         */
+        public Builder folderName(String folder) {
+            this.folder = Environment.getExternalStorageDirectory() + FILE_SEPARATOR + folder;
+            return this;
+        }
+
+        /**
+         * Path to store logs relative to system root. Might not need storage permission.
+         * To use internal app storage call with context.getFilesDir().getAbsolutePath()
+         *
+         * @param folder full folder path
+         */
+        public Builder folderFullPath(String folder) {
+            this.folder = folder;
+            return this;
+        }
+
+        /**
+         * Defaults to comma `,` for CSV, but can be changed here
+         *
+         * @param separator default is `,`
+         */
+        public Builder separator(String separator) {
+            this.separator = separator;
+            return this;
+        }
+
+        /**
+         * Format to pass to SimpleDateFormat for human-readable dates.
+         * See android.icu.text.SimpleDateFormat docs for possible formats
+         *
+         * @param dateFormat defaults to yyyy.MM.dd HH:mm:ss.SSS
+         */
+        public Builder dateFormat(String dateFormat) {
+            this.dateFormat = dateFormat;
+            return this;
+        }
+
+        /**
+         * locale to pass to the SimpleDateFormat for human-readable dates.
+         *
+         * @param locale defaults to locale.UK
+         */
+        public Builder locale(Locale locale) {
+            this.locale = locale;
+            return this;
+        }
+
+        /**
+         * Looper to run the write (I/O) operation
+         *
+         * @param looper defaults to a newly created Loop in a HandlerThread.
+         */
+        public Builder looper(Looper looper) {
+            this.looper = looper;
+            return this;
+        }
+
+        /**
+         * Builds the instance of the FileLogger
+         *
+         * @return a new instance of FileLogger
+         */
+        public FileLogger build() {
+            if (this.looper == null) {
+                HandlerThread ht = new HandlerThread("FileLogger." + folder);
+                ht.start();
+                this.looper = ht.getLooper();
+            }
+            if (this.folder == null) {
+                this.folder = Environment.getExternalStorageDirectory() +
```

Comment:
I agree internal storage is better. But the only way to default to internal storage is to have access to context, which as far as I could see, it's not available anywhere in the library. I didn't want to disrupt the existing API on deeper levels, so a simple way is to let external developers set the folder directly.

And as I mentioned previously. There is a clear use case of having different folders matching code versions.